source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+{% endnote %}
+
+Use `cy.route2()` to manage the behavior of network requests at the network layer.
+
+## Comparison to cy.route()
+
+Unlike {% url `cy.route()` route %}, `cy.route2()`:
+
+- can mock all types of network requests (i.e., Fetch API, page loads, XMLHttpRequests, etc.)
+- no longer requires {% url `cy.server()` server %} before use
+
+# Syntax
+
+```javascript
+cy.route2(requestMatcher)
+cy.route2(requestMatcher, response)
+cy.route2(method, requestMatcher)
+cy.route2(method, requestMatcher, response)
+```
+
+## Usage
+
+**{% fa fa-check-circle green %} Correct Usage**
+
+```javascript
+cy.route2('/users/**')
+```
+
+## Arguments
+
+**{% fa fa-angle-right %} requestMatcher** ***(String, Glob, RegExp, Object)***
+
+The `requestMatcher` argument allows you to listen for a route matching a specific URL or pattern.
+
+For simple route matching, passing a string, glob, or RegExp is the simplest method:
+
+```js
+// String
+cy.route2('/users)
+
+// Regex
+cy.route2(/users\/\d+/)
+
+// Glob to match all paths in the /user path
+cy.route2('/users/**')
+
+// https://localhost:8080/users/1b2c3            <-- matches
+// https://localhost:8080/users/profile/edit  <-- matches
+// https://localhost:8080/users/transaction?month=03&year=2020 <-- matches
+
+```
+
+However, if you need to match additional properties on the route, you can pass an object instead.
+
+```js
+cy.route2({
+  path: '/users/**',
+  method: 'POST',
+})
+```
+
+The following contains a complete list of available properties you can verify against:
+
+Option | Default | Description
+--- | --- | ---
+`auth` | `null` | Object with a `username` and `password` to match HTTP basic authentication
+`headers` | `null` | Object to match client request headers
+`hostname` | `null` | String, RegExp, or Glob to match based on requested hostname
+`method` | `'ALL'` | Match based on HTTP method (`GET`, `POST`, `PUT`, etc.)
+`path` | `null` | String, RegExp, or Glob to match on request path after the hostname, including query params
+`pathname` | `null` | String, RegExp, or Glob to match on request path after the hostname, without query params
+`port` | `null` | Match based on requested port number
+`query` | `null` | Object to match based on query parameters
+`url` | `null` | String, RegExp, or Glob to match based on full request URL
+
+**{% fa fa-angle-right %} response** ***(String, Object, Array, Function)***
+
+This allows you to supply a response `body` to {% url 'stub' stubs-spies-and-clocks#Stubs %} in the matching route. This is accomplished by giving you access to the request that you can use in a plain function.
+
+```js
+cy.route2('/users/**', (req) => {
+  req.send({
+    statusCode: 200,
+    body: JSON.stringify({
+      profile: {
+        firstName: 'Tony',
+        lastName: 'Jarvis'
+      }
+    })
+  })
+})
+```
```

Comment:
This is a strange example, this can be done more directly like this:

```suggestion
```js
cy.route2('/users/**', {
  statusCode: 200,
  body: {
    profile: {
      firstName: 'Tony',
      lastName: 'Jarvis'
  }
})
```

Note that the StaticResponse can be supplied directly, without needing to do the dynamic request/response interception.

Additionally, a `body` that is an `object` will automatically be JSONified, and a `Content-Type: application/json` header will be added to the response automatically. This should probably be mentioned somewhere in this doc. Implementation:

https://github.com/cypress-io/cypress/blob/50a45c1b707ef0ccb0447b94ecb04946e994a4db/packages/driver/src/cy/net-stubbing/static-response-utils.ts#L84-L91